### DISCUSSION TITLE
IP Ban, Geo Blocking, moint point listener timer

### DISCUSSION THREAD

**Discussion by Andy-Beats** [AzuraCast/AzuraCast#5911] [General]
Hello

When is Azuracast going to implement a ban ip & geo block that actually works? 

Currently i have loads of extremely abusive googleusercontent ip's (fake by the look of it) connecting to my server 2 by 2, auto connecting to my station and ripping/scraping my music with extreme force for 20 hours non stop, when i say auto connect i refer to even when i reload and start station., the 2 toxic ip's are immediately connected, i tried banning both ip's on edit profile and shoutcast admin panel, reload and nothing works as the ip's connect again, geo block & reload station, yet the 2 ip's still connect. I then tried to set disconnect listeners after every 60 seconds on mount point just as a one off, saved changes, reloaded and that didn't work either. Seems like Azura is quite far behind in providing a stable up to date working panel all round. Please urgently work on this for improvement.

On a side note, please guys, any advice of what i can do to rid these 2 leeches from my server? Any particular ip format i should be using to ban? What about geoblock not working?

Any advice would be great.

Regards

Andy

**Comment by Andy-Beats** [comment]
Just checked again, a lot is not functioning on Azuracast, geo block not working, ban ip/subnet not working, limit listener connection time not working, stats not accurate. Sooner or later these googleuser bot 24/7 stream scraping abuse will flood their way into more azuracast panels, perhaps it's time to repair the necessary on azuracast, currently it's quite impossible to curate radio without proper stream protection, rest assured.

**Comment by Andy-Beats** [comment]
Seems like not only is Azuracast not functioning properly for broadcasters but also lack of assistance here just the same.

**Comment by BusterNeece** [comment]
Congratulations, you win the "worst poster of the week" award for AzuraCast.

> When is Azuracast going to implement a ban ip & geo block that actually works?

If you had created an Issue on here to help report the bug to us and provide appropriate details, including relevant logs, it's possible we could have it fixed quite quickly. Our understanding is that our IP and Geolocation blocking functionality works (we don't intentionally ship things that don't, after all), and if we find cases where it doesn't, we can only get those resolved once given sufficient information to do so.

> Seems like not only is Azuracast not functioning properly for broadcasters but also lack of assistance here just the same.

After providing your complaint (written, mind you, in the format of a complaint, not a bug report that would help us fix the software), you waited under 24 hours to declare that our support response times are insufficient for you.

AzuraCast is a very complex piece of software maintained by a team that is, on its best days, about two people. On many days, its code is maintained by a single maintainer, me. I provide this software in its entirety for free of charge, and support via GitHub is also free. So your expectation of a one-day turnaround on your *unpaid* support inquiry from the same overworked team of maintainers that gave you the software itself for free is, to put it mildly, unreasonable.

In your anger, it appears you also posted another message that is now deleted. I get any message from GitHub posted directly into my e-mail inbox, however, so I woke up this morning to this message:

> Rounding off a 4th comment, to the seriousness of these queries impacting radio stations so severly on azuracast with dysfunctional facilities obviously not protecting station streams, enough to rather seek centovacast, i am perplexed that no one has answered anything here, are host providers also restricting azuracast functionality to users? Believe me there are some shoddy ego tripping host providers out there with crap immature attitudes one can only match in a trash can.

As a user imposing unreasonable, impatient demands upon a piece of free and open-source software, not understanding how the open-source ecosystem in general works and treating us the way you likely treat the people who make your food and deliver your packages (who all also deserve better), threatening to go to a competitor who is _paid_ to listen to your complaints and provide support, and even then would never, ever fix an issue reported in this manner in a 24-hour timeframe...please. Please go there. Please do not come back.

**Comment by Andy-Beats** [comment]
Hey Buster, you are so kind, seriously, i love the panel but features are lacking, perhaps the worst poster turns out to be the best implement for future azuracast improvements. No anger on my side, far from it, seems like you are, elaborating personal remarks is uncalled for really. All i was referring to is the software.
